### PR TITLE
Issue #DP-125 feat: review flow by sorcing org reviewer/andmin

### PR DESCRIPTION
**Pull request by venkanagouda-k**

### CODE REVIEW

File: src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts [hunk 1]
```diff
@@ -127,6 +129,7 @@ export class ChapterListComponent implements OnInit, OnChanges, OnDestroy, After
         this.uploadHandler(contentMeta);
       }
     };
+    this.sourcingOrgReviewer = this.router.url.includes('/sourcing') ? true : false;
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:132]: can we also check the role here?

Comment (venkanagouda-k) [reply src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:132]: by default, both roles (admin and content_reviewer) become reviewer of TOC
whoever comes from /sourcing, are all reviewer

Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:132]: Just a question to @pallakartheekreddy and @twsvaishali  
what is the correct or better way to check if he is sourcing org user? the one using url or the below code ? 

`isSourcingOrgReviewer () {
    return !!(this.userService.userProfile.userRoles.includes('ORG_ADMIN') ||	    return !!(this.userService.userProfile.userRoles.includes('ORG_ADMIN') ||
    this.userService.userProfile.userRoles.includes('CONTENT_REVIEWER'));	    this.userService.userProfile.userRoles.includes('CONTENT_REVIEWER'));
  }	  }`

File: src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts [hunk 2]
```diff
@@ -255,6 +259,7 @@ export class ChapterListComponent implements OnInit, OnChanges, OnDestroy, After
         instance.countData['awaitingreview'] = 0;
         instance.countData['sampleContenttotal'] = 0;
         instance.countData['sampleMycontribution'] = 0;
+        instance.countData['pendingreview'] = 0;
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:262]: please always use camelCase. Please install `code spell checker` extension on VS Code.

File: src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts (hunk 1 above)
Comment (twsvaishali) [reply src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:132]: Agree with @venkateshwarans bcoz in the future, we dont know who can access the sourcing url

File: src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts [hunk 3]
```diff
@@ -457,9 +467,18 @@ export class ChapterListComponent implements OnInit, OnChanges, OnDestroy, After
   shouldContentBeVisible(content) {
     const creatorViewRole = this.actions.showCreatorView.roles.includes(this.sessionContext.currentRoleId);
     const reviewerViewRole = this.actions.showReviewerView.roles.includes(this.sessionContext.currentRoleId);
-    if (this.userService.userProfile.userRoles.includes('ORG_ADMIN')) {
+    if (this.userService.userProfile.userRoles.includes('ORG_ADMIN') ||
+        this.userService.userProfile.userRoles.includes('CONTENT_REVIEWER')) {
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:471]: There is already a function to check this `isSourcingOrgReviewer()`. Use that.

File: src/app/client/src/app/modules/cbse-program/components/content-uploader/content-uploader.component.ts [hunk 4]
```diff
@@ -670,4 +672,34 @@ export class ContentUploaderComponent implements OnInit, AfterViewInit {
...
+      param: { 'mode': 'edit' }
+    };
+    this.actionService.get(option).pipe(map((res: any) => res.result.content)).subscribe((data) => {
+      const request = {
+        content: {
+        'versionKey': data.versionKey
+        }
+      };
+      const acceptedContents = data.acceptedContents ? data.acceptedContents : [];
+      const rejectedContents = data.rejectedContents ? data.rejectedContents : [];
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/content-uploader/content-uploader.component.ts:689]: Just use `||` instead :)

File: src/app/client/src/app/modules/cbse-program/components/question-list/question-list.component.ts [hunk 5]
```diff
@@ -827,5 +829,33 @@ export class QuestionListComponent implements OnInit, AfterViewInit, OnDestroy {
...
+        this.programStageService.removeLastStage();
+      }, (err) => {
+        action === 'accept' ? this.toasterService.error(this.resourceService.messages.fmsg.m00102) :
+                              this.toasterService.error(this.resourceService.messages.fmsg.m00100);
+      });
+    }, (err) => {
+      action === 'accept' ? this.toasterService.error(this.resourceService.messages.fmsg.m00102) :
+                              this.toasterService.error(this.resourceService.messages.fmsg.m00100);
+    });
+  }
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/cbse-program/components/question-list/question-list.component.ts:860]: Can you move this function into a service? we are using it at two places, right?

File: src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts (hunk 1 above)
Comment (pallakartheekreddy) [reply src/app/client/src/app/modules/cbse-program/components/chapter-list/chapter-list.component.ts:132]: Currently, Sourcing URL can access only ORG_ADMIN & CONTENT_REVIEWER We are already handling in authGurd right? So the above condition looks fine for me..

**Pull request merged**